game_patch/misc/g_solid.cpp

```diff
@@ -409,6 +410,53 @@ void g_solid_render_ui()
     }
 }
 
+std::optional<int> bm_get_cache_slot_opt(int handle)
+{
+    int cache_slot = rf::bm_get_cache_slot(handle);
+    return (cache_slot == -1) ? std::nullopt : std::optional<int>{cache_slot};
+}
+
+int bm_get_width(int handle)
```

Comment:
there's `rf::bm::get_dimensions(bitmap_handle, &bm_w, &bm_h);`

You really through that there would be no such function?
Also I have no idea why you didn't use `BitmapEntry` structure definition and instead put a lot of magic numbers into that code...